### PR TITLE
Allow usage as a rust dependency

### PR DESCRIPTION
**Pull request by caass**
This code is super disgusting i think. please tell me what you'd like me to do to make it less...bad...unless you like it as is!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,37 +212,36 @@ If you maintain a community distribution and want it listed here, leave an issue
...
-    -p, --ports <ports>...           A list of comma separed ports to be scanned. Example: 80,443,8080.
-    -r, --range <range>              A range of ports with format start-end. Example: 1-1000
         --scan-order <scan-order>    The order of scanning to be performed. The "serial" option will scan ports in
                                      ascending order while the "random" option will scan ports randomly [default:
                                      serial]  [possible values: Serial, Random]
     -t, --timeout <timeout>          The timeout in milliseconds before a port is assumed to be closed [default: 1500]
     -u, --ulimit <ulimit>            Automatically ups the ULIMIT with the value you provided
 
 ARGS:
-    <ips-or-hosts>...    A list of comma separated IP addresses or hosts to be scanned
```
Comment (brandonskerritt) [inline README.md:241]: this README is out-dated, it was over-written by a merge conflict. Please update  to the latest one :)

**Review by brandonskerritt — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (caass) [reply README.md]: Done!

**Comment by brandonskerritt** [comment]
This code doesn't compile on our tests :cry: :sparkles:

**Comment by caass** [comment]
wtf...i thought it was just the doctests that were failing to compile, and the real code worked fine? I can fix the docs before merging if you'd like. I almost did that without asking, but it involved `pub`-ing some `fn`s so I didn't want to without your blessing. If that's alright with you I can go ahead and patch up the docs, otherwise I think adding `no_run` or `ignore` or whatever it is will make the builds pass.

**Comment by brandonskerritt** [comment]
> wtf...i thought it was just the doctests that were failing to compile, and the real code worked fine? I can fix the docs before merging if you'd like. I almost did that without asking, but it involved `pub`-ing some `fn`s so I didn't want to without your blessing. If that's alright with you I can go ahead and patch up the docs, otherwise I think adding `no_run` or `ignore` or whatever it is will make the builds pass.

Hey! I'm not really sure what the best method of action here is, as you now know more about Rust libraries than I do :sweat_smile: Whatever you think will make the tests runs will be good :smile:

**Comment by caass** [comment]
ok, no sweat! i'll read up some about how rust generates documentation and runs doc tests. i checked the build logs and i *think* it's just the doctests that are failing, but it's harder to parse than what i get on my machine. i'll push a patch later today ❤️

**Comment by caass** [comment]
re-doing this since i can't be bothered to figure out the merge conflicts. i have the idea now though, so shouldn't be too long!

**Pull request closed without merging**